pkg/scalers/kafka_scaler.go

```diff
@@ -227,7 +227,11 @@ func (s *kafkaScaler) GetMetrics(ctx context.Context, merticName string, metricS
 		totalLag += lag
 	}
 
-	// We should find a way to not scale beyond the number of partitions
+	// don't scale out beyond the number of partitions
+	if (totalLag / s.metadata.lagThreshold) > int64(len(partitions)) {
```

Comment:
I'm not sure I understand this condition. what does the ration between totalLag and lagThreshold have to do with the number of partitions? tbh, I'm not very familiar with Kafka, but I thought the lag is how much behind consumers are from producers on a given partition, right?

Reply:
I'll reach offline for a discussion